cmd/cloudFoundryDeploy.go

```diff
@@ -724,18 +724,26 @@ func handleMtaExtensionCredentials(extFile string, credentials map[string]interf
 	for name, credentialKey := range credentials {
 		credKey, ok := credentialKey.(string)
 		if !ok {
-			return fmt.Errorf("Cannot handle mta extension credentials: Cannot cast '%v' (type %T) to string", credentialKey, credentialKey)
+			return fmt.Errorf("cannot handle mta extension credentials: Cannot cast '%v' (type %T) to string", credentialKey, credentialKey)
 		}
-		pattern := "<%= " + name + " %>"
-		if strings.Contains(content, pattern) {
+
+		const allowedVariableNamePattern = "^[-_A-Za-z0-9]+$"
+		alphaNumOnly := regexp.MustCompile(allowedVariableNamePattern)
+		if !alphaNumOnly.MatchString(name) {
```

Comment:
❗ This is not backward compatible. Up to now there was no such restriction wrt to the variable name. Now we should check since otherwise we might have parts which are interpreted by regex, e.g. "`.*`". I personally think the risk is now. In case the variable naming needs to be adjusted.

Reply:
When we use a string inside a regex we should check that this string does not contain regex pattern, shouldn't we. Somehow like sql injection ...

Reply:
> Leading number allowed?

Yes, why not? We cannot use leading numbers for the environment variables, since they cannot contain leading numbers. But here for the placeholders that should work.

Reply:
BTP: where has been the injection method `<%=\\s*" + name + "\\s*%>` specified. On the SAP Help i c annot find it. is it Piper specific?

Reply:
Don't know actually. Was already like that before I ported it from groovy to go.